src/extensions/agent_modules/shared/common/inc/aduc/adu_module_state.h

```diff
@@ -0,0 +1,30 @@
+
+#ifndef ADU_MODULE_STATE_H
+#define ADU_MODULE_STATE_H
+
+#include <aducpal/time.h> // time_t
+#include <du_agent_sdk/agent_module_interface.h> // ADUC_AGENT_MODULE_INTERFACE
+#include <du_agent_sdk/mqtt_client_settings.h> // for ADUC_MQTT_SETTINGS
+
+
+/**
+ * @brief The module state.
+ */
+typedef struct tagADUC_MQTT_CLIENT_MODULE_STATE
+{
+    void* moduleInitData; //!< Module init data
+    ADUC_MQTT_SETTINGS mqttSettings; //!< MQTT connection settings
+
+    char* mqtt_topic_service2agent; // For v1, we're only subscribe to 1 topic.
```

Comment:
>subscribe [](http://example.com/codeflow?start=58&length=9)

subscribed

Reply:
went with imperative (we only subscribe)